lib/matplotlib/colors.py

```diff
@@ -1368,8 +1368,14 @@ def inverse(self, value):
     def autoscale(self, A):
         """Set *vmin*, *vmax* to min, max of *A*."""
         A = np.asanyarray(A)
-        self.vmin = A.min()
-        self.vmax = A.max()
+        # Ensure that A has non-zero size to compute min/max.
+        # Otherwise put in filler values.
+        if A.size:
```

Comment:
This is hitting _alot_ of use cases as all of the color mappables to through here, would it be possible to do this check in `hexbin` instead?

If we do this here, do we want to use the same logic from the auto limit on the Axis to sure min != max (which I forsee causing divide by 0 errors later)?